pkg/orchestrator/provider/aws/client.go

```diff
@@ -70,34 +70,44 @@ func (c Client) Kind() models.CloudProvider {
 }
 
 // nolint:cyclop
-func (c *Client) DiscoverAssets(ctx context.Context) ([]models.AssetType, error) {
-	regions, err := c.ListAllRegions(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get regions: %w", err)
+func (c *Client) DiscoverAssets(ctx context.Context) provider.AssetDiscoverer {
+	output := make(chan models.AssetType)
+	assetDiscoverer := provider.AssetDiscovererImpl{
+		OutputChan: output,
```

Comment:
Maybe better to create a `CreateAssetDiscovererImpl` func for the creation of the `AssetDiscoverer`?

You can also create it once outside the `DiscoverAssets` func and pass it as an arg to that function, no? (to avoid doing it for each provider implementation)

You might need to add a `SetErr` func to the interface.

Reply:
I don't think it should be created outside of the DiscoverAssets function, it should be down to the provider to decide how to implement the AssetDiscoverer interface. This is just a default naive implementation IMO.

I agree with having a `func NewNaiveAssetDiscoverer() NaiveAssetDiscoverer` function to remove the duplication around the channel creation.

Reply:
make sense